Lib/test/test_dis.py

```diff
@@ -2426,5 +2430,101 @@ def _unroll_caches_as_Instructions(instrs, show_caches=False):
                                   False, None, None, instr.positions)
 
 
+class TestDisCLI(unittest.TestCase):
+
+    def infile(self, content):
+        filename = tempfile.mktemp()
+        self.addCleanup(os_helper.unlink, filename)
+        with open(filename, 'w') as fp:
+            fp.write(content)
+        return filename
+
+    def invoke_dis(self, infile, *flags):
+        output = io.StringIO()
+        with contextlib.redirect_stdout(output):
+            dis.main(args=[*flags, infile])
+        return output.getvalue()
+
+    def check_output(self, source, expect, *flags):
+        with self.subTest(flags):
+            infile = self.infile(source)
+            res = self.invoke_dis(infile, *flags)
+            res = textwrap.dedent(res)
+            expect = textwrap.dedent(expect)
+            self.assertListEqual(res.splitlines(), expect.splitlines())
+
+    def test_invokation(self):
+        # test various combinations of parameters
+        base_flags = [
+            ('-C', '--show-caches'),
+            ('-O', '--show-offsets'),
+            ('-P', '--show-positions'),
+            ('-S', '--specialized'),
+        ]
+
+        infile = self.infile('def f():\n\tprint(x)\n\treturn None')
+
+        for r in range(1, len(base_flags) + 1):
+            for choices in itertools.combinations(base_flags, r=r):
+                for args in itertools.product(*choices):
+                    with self.subTest(args=args[1:]):
+                        _ = self.invoke_dis(infile, *args)
+
+    def test_show_cache(self):
+        # test 'python -m dis -C/--show-caches'
+        source = 'print()'
+        expect = '''\
```

Comment:
I'd use text `wrap.dedent()` so that the expected text can be indented with the code.